x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateConfig.java

```diff
@@ -55,7 +82,7 @@ public String getTemplateName() {
      */
     public byte[] loadBytes() {
         String template = TemplateUtils.loadTemplate(fileName, version,
-            Pattern.quote("${" + versionProperty + "}"));
+            Pattern.quote("${" + versionProperty + "}"), variables);
```

Comment:
I don't like the difference here between how `versionProperty` is treated (that is, wrapped with `${}` and escaped via `Pattern.quote()`) and how the variables are treated (not wrapped with anything and not escaped) - I think this will lead to confusion, and very possibly accidentally ending up with regexes where we expected to have static strings.

If we don't end up escaping the keys in `variables` either here or in `TemplateUtils#replaceVariable`, it should be made clear that the keys will be compiles as regexes.

Reply:
I have pused https://github.com/elastic/elasticsearch/pull/51765/commits/b765370e821caaae07af020a6b54995e673d851c to ensure we wrap and escape everywhere.